cli/src/telemetry/session.rs

```diff
@@ -58,13 +63,20 @@ impl Session {
             ci_name: ci_info::get().name,
         };
 
-        let release_version = get_installed_version()?.to_string();
+        let release_version = get_installed_version()
+            .map_err(|e| ApolloError::from(ErrorDetails::CliInstallError { msg: e.to_string() }))?
```

Comment:
hm, `ApolloError` has `from_source` that keeps track of the original error, but the ergonomics could be improved there. This (and anywhere else we `map_err`) would then be:
```
map_err(|e| ApolloError::from_source(e, ErrorDetails::CliInstallError { msg: e.to_string() }))
```

but then there's also `with_context` on `Result` that we implement in `errors.rs` that should solve this, but that FnOnce doesn't have access to the `E` from `Result<T, E>` -- I tried changing that locally and it's a PITA, so I guess that `from_source` is what we can go with for now.

Reply:
or maybe that doesn't compile? Eh, never mind, bike shed.